code/game/objects/items/weapons/tools/mods/_upgrades.dm

```diff
@@ -286,6 +286,13 @@
 		T.use_fuel_cost *= tool_upgrades[UPGRADE_FUELCOST_MULT]
 	if(tool_upgrades[UPGRADE_POWERCOST_MULT])
 		T.use_power_cost *= tool_upgrades[UPGRADE_POWERCOST_MULT]
+	if(tool_upgrades[UPGRADE_QUALITY])
+		if(T.toggleable)
+			T.switched_on_qualities |= UPGRADE_QUALITY
+			if(T.switched_on)
+				T.tool_qualities |= UPGRADE_QUALITY
+		else
+			T.tool_qualities |= UPGRADE_QUALITY
```

Comment:
I think we also need to reset `switched_on_qualities` and `tool_qualities` to their initial values in `/obj/item/tool/refresh_upgrades()`. Otherwise it will never be removed even if the mod goes away.